packages/firestore/src/model/mutation.ts

```diff
@@ -254,6 +268,16 @@ export abstract class Mutation {
   abstract readonly precondition: Precondition;
 }
 
+/**
+ * Used to represent Mutation classes that have a `fieldTransforms` field,
+ * namely SetMutation and PatchMutation. This intermediate class makes it
+ * easier to access the `fieldTransforms` field directly from a Mutation
+ * object.
+ */
+export abstract class MutationWithTransforms extends Mutation {
```

Comment:
Can this just be an interface? Interfaces are neutral in terms of code size and disappear once transpiled to JS.

I also think this should be called `TransformMutation` (sounds familiar? :)) or something. Before you tackle this take a look at the comment below - I think we might be able to simplify the code if Mutation itself has `fieldTransforms`.